FLOE-204: jshint

jobara opened the pull request:
Added grunt with a jshint task. Linted all of the js files. As part of this work, I removed the predefinedOER.js file and instead am fetching the contents from html fragments.
http://issues.fluidproject.org/browse/FLOE-204

jobara committed:
FLOE-204: Add jshint and linted files

Added a grunt jshint task. As part of the linting, I removed the predefinedOER.js file and am fetching the contents from html fragments instead.

jobara committed:
Merge branch 'master' into FLOE-204

jobara committed:
FLOE-204: Slight cleanup of the gruntfile

cindyli commented on demos/js/launcher.js:
```diff
@@ -67,11 +63,17 @@ var demo = demo || {};
                 options: {
                     databaseName: "{launcher}.options.dbName",
-                    content: "{launcher}.options.content",
+                    // content: "{launcher}.options.content",
```
The commented line should be removed.

cindyli commented on Metadata-LICENSE.txt:
```diff
@@ -0,0 +1,227 @@
+
+Educational Community License, Version 2.0
+*************************************
+Copyright 2013 OCAD University
```
The year should be 2013-2014?

jobara replied:
I don't know. The one in Infusion is just 2011.

colinbdclark replied:
From the Canadian Copyright Law book I have on my desk:

"The year to include in a copyright notice should be the year of first publication, or the year in which substantial revisions to a work occurred. For constantly evolving Web sites, the year in the notice would be updated whenever more than trivial revisions or additions are made to the site. While earlier years can remain as part of the notice, the date of the latest substantial revisions must be included... Alternatively, you could include a range of years (e.g. 1996-1999), starting from the date of the oldest elements and ending with the date of the newest elements in the work."

So it's clear to me that a) the year is the least important part of the copyright statement and b) we have some leeway in terms of how we represent it.

Since I believe we typically use a range in our individual files, I don't see any reason why we shouldn't do so in the repository-wide copyright notice as well (e.g. 2013-2014). If we think this will require too much maintenance, we should leave it as the earliest date of publication (e.g. 2013).

I'll leave it up to you guys to decide which approach to take, but we should apply it consistently across all our assets.

jobara replied:
@colinbdclark when you say all our assets do you mean both the license file and the copyright headers in the js files or do you mean metadata vs infusion?

colinbdclark replied:
I meant specifically between different projects. I was raising the question with you of whether you thought it would be easiest to leave the repo-wide license file as "earliest publication date" or whether you want to switch to a form that is consistent with the headers.

jobara replied:
I'm open to either really. Although it would definitely be easier to not update it every year. I think it's something we'd easily forget.

cindyli commented:
metadata/tests/lib/infusion/utils/js/IoCTestUtils.js has fluid_1_5 as the pass-in argument for the anonymous function. It needs to be changed too?

jobara committed:
FLOE-204: removed commented out code

jobara commented:
@cindyli, metadata/tests/lib/infusion/utils/js/IoCTestUtils.js is part of infusion. I don't think we should be changing that.

cindyli commented:
@jobara, we probably should add a note into metadata/src/lib/infusion/VERSION.md to warn that metadata/tests/lib/infusion needs to be updated accordingly at infusion version change. What do you think?

jobara commented:
@cindyli that makes sense

jobara committed:
FLOE-204: Added a comment to infusion version doc

Added a note that the Infusion testing framework needs to be updated along with the any changes to Infusion itself.

cindyli commented:
Merged @ b95c4c138ecaf07ce1a7e694f2c7829c9762ae6f
